var/spack/repos/builtin/packages/dftbplus/package.py

```diff
@@ -160,3 +233,56 @@ def edit(self, spec, prefix):
             )
 
             mconfig.filter("WITH_DFTD3 := .*", "WITH_DFTD3 := 1")
+
+    def cmake_args(self):
+        args = [
+            self.define_from_variant("WITH_OPENMP", "openmp"),
+            self.define_from_variant("WITH_API", "api"),
+            self.define_from_variant("WITH_ARPACK", "arpack"),
+            self.define_from_variant("WITH_CHIMES", "chimes"),
+            self.define_from_variant("WITH_ELSI", "elsi"),
+            self.define_from_variant("WITH_GPU", "gpu"),
+            self.define_from_variant("WITH_MBD", "mbd"),
+            self.define_from_variant("WITH_MPI", "mpi"),
+            self.define_from_variant("WITH_PLUMED", "plumed"),
+            self.define_from_variant("WITH_POISSON", "poisson"),
+            self.define_from_variant("WITH_PYTHON", "python"),
+            self.define_from_variant("WITH_SDFTD3", "sdftd3"),
+            self.define_from_variant("WITH_SOCKETS", "sockets"),
+            self.define_from_variant("WITH_TBLITE", "tblite"),
+            self.define_from_variant("WITH_TRANSPORT", "transport"),
+            self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
+        ]
+        # SCALAPACK
```

Comment:
Can we add this comment from my MR, maybe this adds a bit more information to the future maintainers.
```suggestion
        # SCALAPACK
        # Note: dftbplus@20.1 uses plural form of the option names
        #       (e.g. -DSCALAPACK_LIBRARIES)
        # but dftbplus@20.2 onwards uses singular
        #       (e.g. -DSCALAPACK_LIBRARY)
        # and plural form is ignored.
        # We set both inorder to be compatible with all versions.
```

Reply:
Sorry for that, only after we have changed to CMake and made the first release, did I realize, that projects are not supposed to use the plural form, which are by convention "reserved" for CMake itself. (Of course, this info is well hidden in the CMake docs...) Do we need to support old version at all? Why not starting with 20.2 or even 21.1? Since then, our CMake interface is quite stable. (I don't see any reason, why somebody wanted to install a DFTB+ version which is more than 2 years old.)

Reply:
The ESL bundle still ships with dftbplus@20.1 :sweat_smile:   https://esl.cecam.org/en/bundle/

Reply:
OK, well spotted. :smile: But probably, that should not be the reference (or are there any plans to create a spack recipe for the ESL-bundle?)